### DISCUSSION TITLE
Option to remove html comments

### DISCUSSION THREAD

**Discussion by cdtut** [withastro/roadmap#747] [Proposal]
### Body

There could be option to treat html comments like javascript style comments and remove them from DOM. It is easier with language tools to toggle html comments but that stays in DOM.

Can also be solved by letting us customize HTML compression such as htmlnano then we can remove comments that way.

**Comment by iplanwebsites** [comment]
That'd be handy, any workaround to remove HMTL comments? I realized I'm building a site with a lot of useless html comments.

**Reply by delucis** [reply]
You can use jsx comments as an alternative: 

```diff
- <!-- comment -->
+ {/* comment */}
```

**Reply by Arecsu** [reply]
> You can use jsx comments as an alternative:
> 
> ```diff
> - 
> + {/* comment */}
> ```

great workaround!

**Reply by shinebayar-g** [reply]
Great workaround! Has anyone figured out if it's possible to generate JSX comments instead of HTML comments in VSCode `CMD + /` shortcut?

**Reply by locmai0808** [reply]
@shinebayar-g can find/replace with regex:
```regex
<!--\s*([\s\S]*?)\s*-->
```

and replace with
```regex
{/* $1 */}
```

**Comment by eblocha** [comment]
I want to mention: Angular relies on HTML comment nodes to find the slots to place dynamic content, and for client hydration. I think removing HTML comments in .astro files would be fine, but it should not delete comment nodes generated by prerendering/SSR.
